kratos/input_output/gid_output.h

```diff
@@ -0,0 +1,635 @@
+        const std::string& rDatafilename,
+        const GiD_PostMode Mode,
+        const MultiFileFlag UseMultipleFilesFlag,
+        const WriteDeformedMeshFlag WriteDeformedFlag,
+        const WriteConditionsFlag WriteConditions
+         );
+
+    ///Destructor.
+    ~GidIO() override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Creates the mesh containers for all different element types.
+     * @note The containers are not filled yet in here!
+     */
+    void SetUpMeshContainers();
+
+    /**
+     * @brief Creates the gauss point containers for all different element types.
+     * @note The containers are not filled yet in here!
+     */
+    virtual void SetUpGaussPointContainers();
+
+
+    /// General GidIO related functions ///
+
+    /**
+     * @brief
+     * @todo To be removed
+     */
+    void ChangeOutputName(const std::string& rDatafilename );
+
+    /**
+     * sets up the file names and opens the result file in case there
+     * is ASCII mode and only one file written
+     */
+    void InitializeResultFile( std::string const& rResultFileName );
+
+    /**
+     * TODO: check whether this is still necessary!
+     */
+    void  CloseResultFile();
+
+    /**
+     * TODO: check whether this is still necessary!
+     */
+    void Flush();
+
+    /**
+     * Turn back information as a string.
+     */
+    std::string Info() const override
+    {
+        return "gid io";
+    }
+
+    /**
+     * Print information about this object.
+     */
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << Info();
+    }
+
+    /**
+     * Print object's data.
+     */
+    void PrintData(std::ostream& rOStream) const override
+    {
+    }
+
+    ///result functions
+    /**
+     * This has to be called for each solution step BEFORE any results
+     * (on nodes and on gauss points) is written
+     * @param SolutionTag the current solution step (i.e. time)
+     * @param conditions_flag states whether results should also be written on conditions
+     */
+    virtual void InitializeResults(
+        const double name,
+        const MeshType& rThisMesh
+        );
+    /**
+     * This has to be called for each solution step AFTER all the results
+     * have been written
+     */
+    void FinalizeResults();
+
+    ///functions for writing nodal results
+
+    ///////////////////////////////////////////////////////////////////////
+    //////                  HISTORICAL DATABASE BLOCK                 /////
+    ///////////////////////////////////////////////////////////////////////
+     /**
+     * writes nodal results for variables of type bool
+     */
+    void WriteNodalResults(
+        Variable<bool> const& rVariable,
```

Comment:
```suggestion
        const Variable<bool> const& rVariable,
```

also in other places

Reply:
I can do that, but the main idea was to just move the code, not to refactor yet

Reply:
Well in the GP-Container you did exactly that :D

Reply:
Maybe...

Reply:
In any case you already have a const, you don't need to add another const

Reply:
ah sorry didn't see it :man_facepalming: 
nevermind, sorry for the noise